dd-java-agent/instrumentation/apache-camel-2.20/src/main/java/org/apache/camel/opentracing/ActiveSpanManager.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.opentracing;
+
+import io.opentracing.Span;
+import org.apache.camel.Exchange;
+
+/** Utility class for managing active spans as a stack associated with an exchange. */
+public final class ActiveSpanManager {
+
+  private static final String ACTIVE_SPAN_PROPERTY = "OpenTracing.activeSpan";
+
+  private ActiveSpanManager() {}
+
+  /**
+   * This method returns the current active span associated with the exchange.
+   *
+   * @param exchange The exchange
+   * @return The current active span, or null if none exists
+   */
+  public static Span getSpan(Exchange exchange) {
+    Holder holder = (Holder) exchange.getProperty(ACTIVE_SPAN_PROPERTY);
+    if (holder != null) {
+      return holder.getSpan();
+    }
+    return null;
+  }
+
+  /**
+   * This method activates the supplied span for the supplied exchange. If an existing span is found
+   * for the exchange, this will be pushed onto a stack.
+   *
+   * @param exchange The exchange
+   * @param span The span
+   */
+  public static void activate(Exchange exchange, Span span) {
```

Comment:
I would try to call `datadog.trace.bootstrap.instrumentation.api.AgentTracer#activateSpan` here to active our span at the same time as Camel actives its span. I don't quite understand when and how we should close the resulting scope.

Reply:
Done.